python/thunder/rdds/series.py

```diff
@@ -56,6 +56,22 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise ValueError("Index must be a non-empty indexable object")
+        if self._index is None:
+            self.populateParametersFromFirstRecord()
+        try:
+            lenMatch = len(value) == len(self._index)
+        except:
+            raise ValueError("Index must be an object with a length")
```

Comment:
Here I think the traditional exception would be `AttributeError`? I like the descriptive message though, so it's worth catching and raising your own exception here like you're doing.